models/ios/logging_global/ios_logging_global.yaml

```diff
@@ -0,0 +1,1071 @@
+            type: dict
+            suboptions:
+              level:           
+                description: Logging severity level (0-7)
+                type: int
+                choices: [0,1,2,3,4,5,6,7]
+              severity:
+                description: Logging severity level
+                type: str
+                choices: ["alerts","critical","debugging","emergencies","errors","informational","notifications","warnings"]
+      count:                
+        description: Count every log message and timestamp last occurance
+        type: bool
+      delimiter:    
+        description: Append delimiter to syslog messages
+        type: str
+        suboptions:
+          tcp:           
+            description: Append delimiter to syslog messages over TCP
+            type: bool
+      discriminator:
+        description: Create or modify a message discriminator
+        type: str
+      dmvpn:          
+        description: DMVPN Configuration
+        type: dict
+        suboptions:
+          rate_limit:           
+            description: rate in messages/minute, default is 600 messages/minute (1-10000)
+            type: int
+      esm:        
+        description: Set ESM filter restrictions
+        type: str
+        suboptions:
+          config:           
+            description: Permit/Deny configuration changes from ESM filters
+            type: bool
+      exception:    
+        description: Limit size of exception flush output (4096-2147483647)
+        type: int
+      facility:     
+        description: Facility parameter for syslog messages
+        type: str
+        choices: ["auth","cron","daemon","kern","local0","local1","local2","local3","local4","local5","local6","local7","lpr","mail","news","sys10","sys11","sys12","sys13","sys14","sys9","syslog","user","uucp"]
+      filter:     
+        description: Specify logging filter
+        type: dict
+        suboptions:
+          url: 
+            description: Filter Uniform Resource Locator
+            type: str
+          order: 
+            description: Order of filter execution
+            type: int
+          args:
+            description: Arguments passed to filter module.
+            type: str
+      history:
+        description: Configure syslog history table
+        type: dict
+        suboptions:
+          level:           
+            description: Logging severity level (0-7)
+            type: int
+            choices: [0,1,2,3,4,5,6,7]
+          size:            
+            description: Set history table size (0-500)
+            type: int
+          severity:
+            description: Logging severity level
+            type: str
+            choices: ["alerts","critical","debugging","emergencies","errors","informational","notifications","warnings"]
+      host:           
+        description: Set syslog server IP address and parameters
+        type: dict
+        suboptions:
+          options:           
```

Comment:
```suggestion
```
remove option and indent the model

Reply:
```
an-ios-02(config)#logging host ?
  Hostname or A.B.C.D  IP address of the syslog server
  ipv6                 Configure IPv6 syslog server

an-ios-02(config)#logging host 10.0.2.15 ?
  discriminator         Specify a message discriminator identifier for this
                        logging session
  filtered              Enable filtered logging
  sequence-num-session  Include session sequence number tag in syslog message
  session-id            Specify syslog message session ID tagging
  transport             Specify the transport protocol (default=UDP)
  vrf                   Set VRF option
  xml                   Enable logging in XML
  <cr>

an-ios-02(config)#logging host ipv       
an-ios-02(config)#logging host ipv6 2001:db8:3c4d:15::1a2f:1a2b ?
  discriminator         Specify a message discriminator identifier for this
                        logging session
  filtered              Enable filtered logging
  sequence-num-session  Include session sequence number tag in syslog message
  session-id            Specify syslog message session ID tagging
  transport             Specify the transport protocol (default=UDP)
  vrf                   Set VRF option
  xml                   Enable logging in XML
  <cr>

```
```
Example 1
- name: host Configuration nomal
  cisco.ios.ios_logging_global:
    config:
      host: 
        hostname: 10.0.2.15 #as the command would be logging host 10.0.2.15
      host: 
        ipv6:
          name: 2001:db8:3c4d:15::1a2f:1a2b #as the command would be logging host logging host ipv6 2001:db8:3c4d:15::1a2f:1a2b
```
```
Example 2
- name: host Configuration with suboptions
  cisco.ios.ios_logging_global:
    config:
      host: 
        hostname: 10.0.2.15
      options:
        discriminator: Test #logging host 10.0.2.15 discriminator Test       
```

options  is  same for  ipv6d and  ipv4  
@justjais  what would be a better way to handle this?

Reply:
```suggestion
          host:
            address: 10.0.2.15
            discriminator: test
```

Reply:
^^ This is how u should handle such scenarios and you'll find similar config at many places and resources

Reply:
yes, this is a much simpler way to represent, I was not sure how do I make multiple required statements among a linear structure. hence I went for the other approach! like we don't need a hostname tag for ipv4 but we need ipv6 tag while representing the same, so hostname or ipv6 one of them is required. Or we can rely on the user to be more prominent!